### PR TITLE
sinks add redis plugin

### PR DESCRIPTION
**Pull request by feng-crazy**
sinks add redis plugin

### CODE REVIEW

File: etc/sinks/redis.json [hunk 1]
```diff
@@ -0,0 +1,133 @@
+{
+	"about": {
+		"trial": false,
+		"author": {
+			"name": "DengFeng HE",
+			"email": "contact@emqx.io",
```
Comment (ngjaying) [inline etc/sinks/redis.json:6]: Could you update with your own email, company and website or just leave them empty

File: extensions/sinks/redis/redis.go [hunk 2]
```diff
@@ -0,0 +1,209 @@
+package main
```
Comment (ngjaying) [inline extensions/sinks/redis/redis.go:1]: Please add copyright statement at the top. Check other source file for reference, and sign with your name or email.

File: extensions/sinks/redis/redis.go [hunk 3]
```diff
@@ -0,0 +1,209 @@
...
+				err := r.cli.Set(ctx, r.key, v, r.expiration*time.Second).Err()
+				if err != nil {
+					logger.Error(err)
+					return err
+				}
+				logger.Debugf("send redis string success, key:%s data: %s", r.key, string(v))
+			}
+
+		} else {
+			logger.Debug("not config data key for redis")
```
Comment (ngjaying) [inline extensions/sinks/redis/redis.go:192]: Is it better to validate this in configure?

File: extensions/sinks/redis/redis.go [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+		}
+
+		logger.Debug("insert success", string(v))
+	} else {
+		logger.Debug("insert failed data not is []byte data:", data)
```
Comment (ngjaying) [inline extensions/sinks/redis/redis.go:197]: insert failed, data is not []byte

**Review by ngjaying — CHANGES_REQUESTED** [review]
Great work! Appreciate your contribution. As a request from LF Edge, please sign off your commits. Click "detail" in the DCO check for how to do that. Also, could you please add a doc to docs/plugins/sinks for Chinese and/or English.

File: etc/sinks/redis.json (hunk 1 above)
Comment (feng-crazy) [reply etc/sinks/redis.json:6]: ok

File: extensions/sinks/redis/redis.go (hunk 2 above)
Comment (feng-crazy) [reply extensions/sinks/redis/redis.go:1]: ok

File: extensions/sinks/redis/redis.go (hunk 3 above)
Comment (feng-crazy) [reply extensions/sinks/redis/redis.go:192]: In redis JSON, the key is not optional, These are just records

File: extensions/sinks/redis/redis.go (hunk 4 above)
Comment (feng-crazy) [reply extensions/sinks/redis/redis.go:197]: Only JSON character streams are supported

Comment (ngjaying) [reply extensions/sinks/redis/redis.go:196]: yeah, I mean there is a typo "not is" -> "is not"

**Comment by ngjaying** [comment]
DCO fails with invalid email. You'll need to setup the email
```
$ git config --global user.email name@example.com
```

**Comment by ngjaying** [comment]
@feng-crazy We would like to add this to 1.3.0 release. Are you able to sign off before next Wednesday? Thanks.

**Comment by ngjaying** [comment]
@feng-crazy Thank you again for contribute this plugin. I squash your commits into one and format the user/email to pass the DCO. I force push the updates to the master branch of your fork. Sorry for any inconvenience.

**Review by ngjaying — APPROVED** [review]

**Pull request merged**